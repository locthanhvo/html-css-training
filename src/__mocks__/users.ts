import { STATUS } from '@/constants';

export const USERS = Array.from({ length: 5 }).map((_, index) => ({
  id: `${index}`,
  firstName: 'Devon',
  lastName: 'Lane',
  email: 'devon@mail.com',
  phone: '+237 6 99 88 77 66',
  status: STATUS.VALID,
  createdAt: '2024-06-05 14:46:58',
  updatedAt: '2024-06-05 14:46:58',
}));
