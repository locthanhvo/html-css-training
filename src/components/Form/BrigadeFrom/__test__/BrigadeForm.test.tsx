import { render, screen } from '@testing-library/react';

// Component
import BrigadeForm from '..';
import { BrowserRouter } from 'react-router-dom';
import { USERS } from '@/__mocks__';

const mockOnSubmit = jest.fn();

describe('BrigadeForm components', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('match snapshot with add form', () => {
    const { container } = render(
      <BrowserRouter>
        <BrigadeForm onSubmit={mockOnSubmit} />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it('match snapshot with edit form', () => {
    const { container } = render(
      <BrowserRouter>
        <BrigadeForm
          title="Edit Brigade"
          isEdit
          initialValues={USERS[0]}
          onSubmit={mockOnSubmit}
        />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders Add form by default', () => {
    render(
      <BrowserRouter>
        <BrigadeForm onSubmit={mockOnSubmit} />
      </BrowserRouter>,
    );

    expect(screen.getByRole('button', { name: 'Create' })).toBeInTheDocument();
  });

  it('renders Edit form by default', () => {
    render(
      <BrowserRouter>
        <BrigadeForm title="Edit Brigade" isEdit onSubmit={mockOnSubmit} />
      </BrowserRouter>,
    );

    expect(screen.getByRole('button', { name: 'Update' })).toBeInTheDocument();
  });
});
