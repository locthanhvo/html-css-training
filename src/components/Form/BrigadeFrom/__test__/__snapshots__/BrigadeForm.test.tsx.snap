// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrigadeForm components match snapshot with add form 1`] = `
<div>
  <form
    class="css-1urdh7u"
  >
    <p
      class="chakra-text css-5fbnf6"
    >
      Add Brigade
    </p>
    <div
      class="css-1pyx7hi"
    >
      <div
        aria-label="name"
        class="chakra-form-control css-0"
        name="name"
        placeholder="Please enter name"
        role="group"
        value=""
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r0:"
          id="field-:r0:-label"
        >
          Name
        </label>
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            aria-label="name"
            class="chakra-input css-adz8kx"
            id="field-:r0:"
            name="name"
            placeholder="Please enter name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r1:"
          id="field-:r1:-label"
        >
          Description
        </label>
        <textarea
          aria-label="name"
          class="chakra-textarea css-mie5s1"
          id="field-:r1:"
          name="description"
          placeholder="Please enter description..."
        />
      </div>
      <div
        aria-label="commissioner"
        class="chakra-form-control css-0"
        name="commissioner"
        placeholder="Please enter commissioner"
        role="group"
        value=""
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r2:"
          id="field-:r2:-label"
        >
          Commissioner
        </label>
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            aria-label="commissioner"
            class="chakra-input css-adz8kx"
            id="field-:r2:"
            name="commissioner"
            placeholder="Please enter commissioner"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="css-j7qwjs"
      >
        <p
          class="chakra-text css-1hinmgu"
        >
          Role
        </p>
        <button
          aria-controls="menu-list-:r4:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-14e585h"
          id="menu-button-:r4:"
          type="button"
        >
          <span
            class="css-xl71ch"
          >
            Admin
          </span>
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              fill="none"
              height="7"
              viewBox="0 0 12 7"
              width="12"
            >
              <path
                d="M11 1L6 6L1 1"
                stroke="black"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </button>
        <div
          class="css-0"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-sjt5nk"
            id="menu-list-:r4:"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-kjnj8a"
              data-index="0"
              id="menu-list-:r4:-menuitem-:r5:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Admin
            </button>
            <button
              class="chakra-menu__menuitem css-wntb6v"
              data-index="1"
              id="menu-list-:r4:-menuitem-:r6:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Member
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-bbgbcw"
    >
      <button
        class="chakra-button css-ey593w"
        disabled=""
        name="Create"
        type="submit"
      >
        Create
      </button>
      <button
        class="chakra-button css-15qijou"
        name="Cancel"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;

exports[`BrigadeForm components match snapshot with edit form 1`] = `
<div>
  <form
    class="css-1urdh7u"
  >
    <p
      class="chakra-text css-5fbnf6"
    >
      Edit Brigade
    </p>
    <div
      class="css-1pyx7hi"
    >
      <div
        aria-label="name"
        class="chakra-form-control css-0"
        name="name"
        placeholder="Please enter name"
        role="group"
        value=""
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r7:"
          id="field-:r7:-label"
        >
          Name
        </label>
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            aria-label="name"
            class="chakra-input css-adz8kx"
            id="field-:r7:"
            name="name"
            placeholder="Please enter name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r8:"
          id="field-:r8:-label"
        >
          Description
        </label>
        <textarea
          aria-label="name"
          class="chakra-textarea css-mie5s1"
          id="field-:r8:"
          name="description"
          placeholder="Please enter description..."
        />
      </div>
      <div
        aria-label="commissioner"
        class="chakra-form-control css-0"
        name="commissioner"
        placeholder="Please enter commissioner"
        role="group"
        value=""
      >
        <label
          class="chakra-form__label css-zecyy3"
          for="field-:r9:"
          id="field-:r9:-label"
        >
          Commissioner
        </label>
        <div
          class="chakra-input__group css-bx0blc"
          data-group="true"
        >
          <input
            aria-label="commissioner"
            class="chakra-input css-adz8kx"
            id="field-:r9:"
            name="commissioner"
            placeholder="Please enter commissioner"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="css-j7qwjs"
      >
        <p
          class="chakra-text css-1hinmgu"
        >
          Role
        </p>
        <button
          aria-controls="menu-list-:rb:"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-14e585h"
          id="menu-button-:rb:"
          type="button"
        >
          <span
            class="css-xl71ch"
          >
            Admin
          </span>
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              fill="none"
              height="7"
              viewBox="0 0 12 7"
              width="12"
            >
              <path
                d="M11 1L6 6L1 1"
                stroke="black"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </button>
        <div
          class="css-0"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-sjt5nk"
            id="menu-list-:rb:"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-kjnj8a"
              data-index="0"
              id="menu-list-:rb:-menuitem-:rc:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Admin
            </button>
            <button
              class="chakra-menu__menuitem css-wntb6v"
              data-index="1"
              id="menu-list-:rb:-menuitem-:rd:"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              Member
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-bbgbcw"
    >
      <button
        class="chakra-button css-ey593w"
        disabled=""
        name="Update"
        type="submit"
      >
        Update
      </button>
      <button
        class="chakra-button css-15qijou"
        name="Cancel"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</div>
`;
